third_party/meerkat/Component/mmDiscovery/service_server.cpp

```diff
@@ -110,18 +110,20 @@ VOID CServiceServer::DataRecv(OSAL_Socket_Handle iEventSock,
     std::vector<char*> argv;
     t_HandlePacket(argv, pData + strlen(kServiceRequestScheme));
 
-    char server_address[35] = {'\0',};
-    snprintf(server_address, sizeof(server_address) - 1,
-             "--enable-castanets=%s", pszsource_addr);
-    argv.push_back(server_address);
-
-    // TODO: |server_address_old| should be remove after applying
-    // https://github.com/Samsung/Castanets/pull/75.
-    char server_address_old[35] = {'\0',};
-    snprintf(server_address_old, sizeof(server_address_old) - 1,
-             "--server-address=%s", pszsource_addr);
-    argv.push_back(server_address_old);
-
+    const std::string switches(argv.front());
+    if (!switches.find("--enable-service-offloading")) {
```

Comment:
Why don't you check the condition like the below?
if (switches.find("--enable-castanets"))

Reply:
The original code intended to add the --enable-castanets when it is not passed by 'ServiceRequest'. That's why following lines are used:

      char server_address[35] = {'\0',};
      snprintf(server_address, sizeof(server_address) - 1,
               "--enable-castanets=%s", pszsource_addr);
      argv.push_back(server_address);

Reply:
This is arguments from resource offloading and there's no --enable-castanets but they need it.

I/CHROMIUM( 3553): [INFO:WRT(0)] app.js RequestService : castanets, android, com.samsung.android.castanets --type=offloadworker --signaling-server=http://192.168.0.103:9559/offload-js, 192.168.0.96

Reply:
Do you mean resource offloading needs --enable-castanets flag?

Reply:
no, @younghajung told me only render offloading uses the flag. we can use --enable-castanets flag in the if condition.